### PR TITLE
build: Use pkg-config in BITCOIN_QT_CONFIGURE for all hosts including Windows

### PR DESCRIPTION
**Pull request by hebasto**
This PR makes `bitcoin_qt.m4` to use `pkg-config` for all hosts and removes non-pkg-config paths from it. This is a step towards the idea which was clear [stated](https://github.com/bitcoin/bitcoin/pull/8314#issue-76644643) by Cory Fields:
> I believe the consensus is to treat Windows like the others and require pkg-config across the board. We can drop all of the non-pkg-config paths, and simply AC_REQUIRE(PKG_PROG_PKG_CONFIG)

There are two unsolved problems with this PR. If depends is built with `DEBUG=1` the `configure` script fails to pickup Qt:
- for macOS host (similar to, but not the same as #16391)
- for Windows host (regression)

The fix is ~on its way~ submitted in #18298 (as a followup).

Also this PR picks some small improvements from #17820.

### CODE REVIEW

**Comment by hebasto** [comment]
May I ask for gitian builds?

**Comment by hebasto** [comment]
Updated f71b56047bc319184c1f771e0757faeff26cf242 -> e7cfcf79ea9992e2b844ffdcd64ee02af4f9fc07 ([pr18297.01](https://github.com/hebasto/bitcoin/commits/pr18297.01) -> [pr18297.02](https://github.com/hebasto/bitcoin/commits/pr18297.02), [compare](https://github.com/hebasto/bitcoin/compare/pr18297.01..pr18297.02)):

- removed redundant `include` and `lib` tracking for MinGW in `config.site`

**Comment by hebasto** [comment]
Updated e7cfcf79ea9992e2b844ffdcd64ee02af4f9fc07 -> 8a26848c460160e1279f26bc413f693a34e33b9d ([pr18297.02](https://github.com/hebasto/bitcoin/commits/pr18297.02) -> [pr18297.03](https://github.com/hebasto/bitcoin/commits/pr18297.03), [diff](https://github.com/hebasto/bitcoin/compare/pr18297.02..pr18297.03)):

- removed unused now internal functions `_BITCOIN_QT_CHECK_QT5` and `_BITCOIN_QT_CHECK_QT58`

**Comment by hebasto** [comment]
Just noted that some dead code remains:https://github.com/bitcoin/bitcoin/blob/8a26848c460160e1279f26bc413f693a34e33b9d/build-aux/m4/bitcoin_qt.m4#L299-L301

The `bitcoin_cv_qt58` variable is always unset (see #17821).

This code path runs for static Qt builds, including gitian builds. On xenial with Qt 5.5.1 (the minimal supported version) builds against system libs are ok (tested locally + Travis).

Going to remove this dead code in a followup or with other changes requested by reviewers.

**Comment by laanwj** [comment]
Code review ACK 8a26848c460160e1279f26bc413f693a34e33b9d

**Comment by theuni** [comment]
@laanwj Thanks for the ping, will review thoroughly in the next few days.

File: depends/patches/qt/fix_qt_pkgconfig.patch [hunk 1]
```diff
@@ -1,11 +1,23 @@
 --- old/qtbase/mkspecs/features/qt_module.prf
 +++ new/qtbase/mkspecs/features/qt_module.prf
-@@ -245,7 +245,7 @@
+@@ -264,7 +264,7 @@
  load(qt_targets)
  
  # this builds on top of qt_common
 -!internal_module:!lib_bundle:if(unix|mingw) {
-+unix|mingw {
```
Comment (dongcarl) [inline depends/patches/qt/fix_qt_pkgconfig.patch:8]: Do you or @jonasschnelli or @theuni understand why this original patch was necessary?

**Review by dongcarl — COMMENTED** [review]
Code Review ACK 8a26848c460160e1279f26bc413f693a34e33b9d

File: depends/patches/qt/fix_qt_pkgconfig.patch (hunk 1 above)
Comment (theuni) [reply depends/patches/qt/fix_qt_pkgconfig.patch:8]: @dongcarl I don't recall, but from a quick glance, it looks like it was necessary to get qt to generate pkg-config files for internal/bundled libs (png/harfbuzz/pcre, etc).

**Review by theuni — APPROVED** [review]
Code review ACK 8a26848c460160e1279f26bc413f693a34e33b9d

**Pull request merged**